File: content/03/code/ruby/07-add-person-parameter/features/step_definitions/steps.rb
require 'shouty'

Given("{person} is located/standing {int} metre(s) from {person}") do |lucy, distance, sean|
  @lucy = lucy
  @sean = sean
  lucy.move_to(distance)
end

When("Sean shouts {string}") do |message|
  @sean.shout(message)
  @message_from_sean = message
end

Then("Lucy hears Sean's message") do
  expect(@lucy.messages_heard).to eq [@message_from_sean]
end

Then("Lucy does not hear Sean's message") do
  expect(@lucy.messages_heard).not.to include [@message_from_sean]
end

File: features/step_definitions/steps.js
const { Given, When, Then } = require('@cucumber/cucumber')

Given('Lucy is located {int} metres from Sean', function (distance) {
  this.lucy = new Person
  this.sean = new Person
  this.lucy.moveTo(distance)
})

When('Sean shouts {string}', function (string) {
  // Write code here that turns the phrase above into concrete actions
  return 'pending'
})

Then('Lucy hears Sean\'s message', function () {
  // Write code here that turns the phrase above into concrete actions
  return 'pending'
})


